src/controllers/secret.go

```diff
@@ -0,0 +1,205 @@
+package controllers
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	"github.com/dana-team/platform-backend/src/types"
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+)
+
+type SecretController interface {
+	// CreateSecret creates a new secret in the specified namespace.
+	CreateSecret(namespace string, request types.CreateSecretRequest) (types.CreateSecretResponse, error)
+
+	// GetSecrets gets all secretes from the specified namespace.
+	GetSecrets(namespace string) (types.GetSecretsResponse, error)
+
+	// GetSecret gets a specific secret from the specified namespace.
+	GetSecret(namespace string, name string) (types.GetSecretResponse, error)
+
+	// PatchSecret patches a specific secret in the specified namespace.
+	PatchSecret(namespace string, name string, request types.PatchSecretRequest) (types.PatchSecretResponse, error)
+
+	// DeleteSecret deletes a specific secret in the specified namespace.
+	DeleteSecret(namespace string, name string) (types.DeleteSecretResponse, error)
+}
+
+type secretController struct {
+	client kubernetes.Interface
+	ctx    context.Context
+	logger *zap.Logger
+}
+
+// NewSecretController creates a new secrets controller to make requests of
+// CRUD operations using Kubernetes API.
+func NewSecretController(client kubernetes.Interface, context context.Context, logger *zap.Logger) SecretController {
+	return &secretController{
+		client: client,
+		ctx:    context,
+		logger: logger,
+	}
+}
+
+func (n *secretController) CreateSecret(namespace string, request types.CreateSecretRequest) (types.CreateSecretResponse, error) {
```

Comment:
Creates GoDoc strings for all the controller function

Reply:
We already have comments in interface, because comments in the implemented function is not displayed to the user.